hera_cal/io.py

```diff
@@ -0,0 +1,286 @@
+    
+    if pop_autos:
+        for bl in data.keys():
+            if bl[0] == bl[1]:
+                del data[bl], flags[bl]
+
+    # If we don't want nested dicts, convert to DataContainer
+    if not nested_dict:
+        data, flags = DataContainer(data), DataContainer(flags)
+
+    # get meta
+    if return_meta:
+        freqs = np.unique(uvd.freq_array)
+        times = np.unique(uvd.time_array)
+        lsts = np.unique(uvd.lst_array)
+        antpos, ants = uvd.get_ENU_antpos(center=True, pick_data_ants=pick_data_ants)
+        antpos = odict(zip(ants, antpos))
+        pols = np.array([polnum2str(polnum) for polnum in uvd.polarization_array])
+        return data, flags, antpos, ants, freqs, times, lsts, pols
+    else:
+        return data, flags
+
+
+def write_vis(outfilename, data, flags, format='miriad', history='', clobber=False, **kwargs):
+    '''TODO: migrate in hera_cal.utils.data_to_miriad and generalize to also write uvfits.'''
+    raise NotImplementedError('This function has not been implemented yet.')
+
+
+def update_vis(infilename, outfilename, format_in='miriad', format_out='miriad', 
+               data=None, flags=None, add_to_history='', clobber=False, **kwargs):
+    '''Loads an existing file with pyuvdata, modifies some subset of of its parameters, and 
+    then writes a new file to disk. Cannot modify the shape of data arrays. More than one 
+    spectral window is not supported.
+
+    Arguments:
+        infilename: filename of the base visibility file to be updated, or UVData object
+        outfilename: filename of the new visibility file
+        format_in: either 'miriad' or 'uvfits' (ignored if infile is a UVData object)
+        format_out: either 'miriad' or 'uvfits'
+        data: dictionary or DataContainer of complex visibility data to update. Keys
+            like (0,1,'xx') and shape=(Ntimes,Nfreqs). Default (None) does not update. 
+        flags: dictionary or DataContainer of data flags to update. 
+            Default (None) does not update.
+        add_to_history: prepends a string to the history of the output file
+        clobber: if True, overwrites existing file at outfilename
+        kwargs: dictionary mapping updated attributs to their new values. 
+            See pyuvdata.UVData documentation for more info.
+    '''
+
+    # Load infile
+    if type(infilename) == UVData:
+        uvd = deepcopy(infilename)
+    else:    
+        uvd = UVData()
+        if format_in is 'miriad':
+            uvd.read_miriad(infilename)
+        elif format_in is 'uvfits':
+            #TODO: implement this
+            raise NotImplementedError('This function has not been implemented yet.')
+        else:
+            raise TypeError("Input format must be either 'miriad' or 'uvfits'.")
+
+    # set data and/or flags
+    if data is not None or flags is not None:
+        bls = map(uvd.baseline_to_antnums, np.unique(uvd.baseline_array))
```

Comment:
I think I want to force the user to provide a data container that has all the same keys as the file. I don't want to produce a file with mixed sources of data. 

Honestly, I don't think we should ever be popping autos, but I included that as legacy support for Nick (but made the default False rather than True as he had it).